shards: introduce loaded.ready field

keegancsmith opened the pull request:
If this field is false we will communicate back to the client that we
are partially available. We piggy-back on the Crashes stat since that is
currently used by Sourcegraph to communicate back partial availability.
However, once we stabilize here we should introduce more fine grained
fields.

This commit just introduces reads to the field, the next commit will
introduce the logic which mutates this field which will actually change
behaviour.

Test Plan: go test

plz-review-url: https://plz.review/review/15656

stefanhengl commented on shards/shards.go:
```diff
@@ -180,6 +180,10 @@ type loaded struct {
 	// should not be mutated.
 	shards []*rankedShard
+
+	// ready is true if sharded searcher has finisihed loading all initial
```
```suggestion
	// ready is true if sharded searcher has finished loading all initial
```

plz-review[bot] committed:
shards: introduce loaded.ready field

If this field is false we will communicate back to the client that we
are partially available. We piggy-back on the Crashes stat since that is
currently used by Sourcegraph to communicate back partial availability.
However, once we stabilize here we should introduce more fine grained
fields.

This commit just introduces reads to the field, the next commit will
introduce the logic which mutates this field which will actually change
behaviour.

Test Plan: go test

plz-review-url: https://plz.review/review/15656

keegancsmith committed:
Update shards/shards.go

Co-authored-by: Stefan Hengl <stefan@sourcegraph.com>
